cmake.md

```diff
@@ -0,0 +1,30 @@
+## CMake build system
+### Terminology
+
+The CMake system in symbiflow-arch-defs uses the following hierarchy:
+
+* ARCH - The architecture of FPGA.  This is used to capture architecture level parameters (e.g. tech libraries).
+* DEVICE_TYPE - A type of device.  This is defined as an ARCH and the arch.xml which defines the units with this device type.
+* DEVICE - A specific instance of a DEVICE_TYPE.  A device definition is a DEVICE_TYPE, a PACKAGE list, and IO map definitions for each package if needed.
+* BOARD - A specific DEVICE and PACKAGE instance, along with a PROG_TOOL/PROG_CMD.  A board is what is targeted by ADD_FPGA_TARGET.
+
+### Creating a new FPGA target
```

Comment:
I'm not sure what "creating a new FPGA target" actually means? 

I think this means adding a new test (which will run pnr for an FPGA)?